### PR TITLE
feat(Core/Player): allow to queue both RDF & BGs together

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10099

## Tests Performed:

- Not Tested
- 

## How to Test the Changes:

1. https://github.com/azerothcore/azerothcore-wotlk/issues/10099
2.
3.

## Known Issues and TODO List:

- [ ] My english is shit So please fix description in worldserver.conf.dist ^^
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by acidmanifesto — APPROVED** [review]
Description is fine with me.

**Comment by Nyeriah** [comment]
What happens if they accept the BG call during the LFG role check proposal?
Also what happens if they are inside a BG and get a LFG call, vice versa?
Might want to remove them from the other queue once they accept one of the queues?

File: src/server/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -3622,6 +3622,14 @@ LFG.KickPreventionTimer = 900
 
 EnablePlayerSettings = 0
 
+#
+#     Allow.Join.BG.And.LFG
+#        Description: Allow join BG and LFG in the same time.
```
Comment (Nyeriah) [inline src/server/worldserver/worldserver.conf.dist:3627]:
```suggestion
#        Description: Allow queueing for BG and LFG at the same time.
```

**Comment by FrancescoBorzi** [comment]
@IntelligentQuantum thanks for this PR in record time! :) did you handle cases like:

- the player is already in BG and asked to join RDF
- the player is already in RDF and asked to join BG

or does the other queue cancel as soon as you join one of them?

**Comment by acidmanifesto** [comment]
> What happens if they accept the BG call during the LFG role check proposal?
> Also what happens if they are inside a BG and get a LFG call, vice versa?
> Might want to remove them from the other queue once they accept one of the queues?

I know and can only speak for sl retail in this case. On sl retail you will remain in queue for the other while in the other and if u accept u will be in the group/bg with a deserter aura. Blizzlike at its finest

**Comment by Nyeriah** [comment]
> > What happens if they accept the BG call during the LFG role check proposal?
> > Also what happens if they are inside a BG and get a LFG call, vice versa?
> > Might want to remove them from the other queue once they accept one of the queues?
> 
> I know and can only speak for sl retail in this case. On sl retail you will remain in queue for the other while in the other and if u accept u will be in the group/bg with a deserter aura. Blizzlike at its finest

Yes, that's not exactly what I meant though. It was more in the sense of, does the queue still work if that happens. The RDF queue is prone to jamming from what I can tell, so we must ensure it doesn't break once they queue to multiple things.

But I think it should be fine

**Comment by temperrr** [comment]
It does exactly what we described.

You can queue for both, whichever pops up first you get queued into, then removes the other queues.

Do we want to extend the description to let players stay in queue as on hold whilst they do a battleground and vice versa?

File: src/server/worldserver/worldserver.conf.dist [hunk 2]
```diff
@@ -3622,6 +3622,14 @@ LFG.KickPreventionTimer = 900
 
 EnablePlayerSettings = 0
 
+#
+#     Allow.Join.BG.And.LFG
+#        Description: Allow queueing for BG and LFG at the same time.
+#        Default:     0 - Disabled
+#                     1 - Enabled
+
+Allow.Join.BG.And.LFG = 0
```
Comment (Winfidonarleyan) [inline src/server/worldserver/worldserver.conf.dist:3631]: `JoinBGAndLFG.Enable = 0`

Comment (Nyeriah) [reply src/server/worldserver/worldserver.conf.dist:3631]: @IntelligentQuantum could you check this one please?

**Comment by Nyeriah** [comment]
There's unexpected logic here

Skärmbild 2022-01-25 094658

If the setting is **disabled**, it will fall into that IF and leave the conditional segment. In other words, if the setting is disabled, no other RDF requirement is ever checked, leading level requirements to be bypassed among other things.

This breaks the RDF queue. Could you check this @IntelligentQuantum ?

**Pull request merged**